tests/posix/common/src/pthread.c

```diff
@@ -465,6 +465,50 @@ ZTEST(pthread, test_pthread_cleanup)
 	zassert_ok(pthread_join(th, NULL));
 }
 
+static void *test_pthread_cancel_fn(void *arg)
+{
+	int oldstate = ~PTHREAD_CANCEL_ENABLE;
+
+	zassert_ok(pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, &oldstate));
+	zassert_equal(oldstate, PTHREAD_CANCEL_ENABLE);
```

Comment:
This test doesn't really care about or need its previous `oldstate`

```suggestion
	zassert_ok(pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, NULL));
```